x-pack/plugins/lens/public/embeddable/embeddable.tsx

```diff
@@ -610,6 +611,38 @@ export class Embeddable
 
   private fullAttributes: LensSavedObjectAttributes | undefined;
 
+  private handleExternalUserMessage = (messages: UserMessage[]) => {
+    if (this.input.customBadgeMessages) {
+      // we need something else to better identify those errors
+      const { messagesToHandle, originalMessages } = messages.reduce<{
+        messagesToHandle: UserMessage[];
+        originalMessages: UserMessage[];
+      }>(
+        (acc, message) => {
+          const hasEmbeddableBadge = message.displayLocations.some(
+            (d) => d.id === 'embeddableBadge'
+          );
+
+          if (hasEmbeddableBadge) {
+            acc.messagesToHandle.push(message);
+          } else {
+            acc.originalMessages.push(message);
+          }
+
+          return acc;
+        },
+        { messagesToHandle: [], originalMessages: [] }
+      );
```

Comment:
nit: I think this can use `partition` from lodash to be more readable:

```suggestion
      const [ messagesToHandle, originalMessages] = partition(
        messages,
        message => message.displayLocations.includes('embeddableBadge')
      );
```